### ISSUE TITLE
Errors with Specialization Modules in 2.03

### ISSUE THREAD

**Issue by Vegeir** [MilleXIV/XIV_Databar#42]
Using Version 2.03 as 100 Fire Mage with bar set to TOP position.  2 errors related to Spec. Module:

**1. Error**: _Left-clicking on spec immediately caused this occur:_

```
12x XIV_Databar\modules\talent-Talent.lua:296: attempt to index field 'color' (a nil value)
XIV_Databar\modules\talent-Talent.lua:296: in function <XIV_Databar\modules\talent.lua:295>
Locals:
(*temporary) = <function> defined =[C]:-1
(*temporary) = <unnamed> {
0 = <userdata>
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index field 'color' (a nil value)"
buttonText = <unnamed> {
0 = <userdata>
}
db = <table> {
general = <table> {
}
modules = <table> {
}
}
```

**2. Error**: _Right-clicking on Spec button (to select loot choice) also causes errors.  It looks like the same function being called at a different line number:_

```
14x XIV_Databar\modules\talent-Talent.lua:390: attempt to index field 'color' (a nil value)
XIV_Databar\modules\talent-Talent.lua:390: in function <XIV_Databar\modules\talent.lua:389>
Locals:
(*temporary) = <function> defined =[C]:-1
(*temporary) = <unnamed> {
0 = <userdata>
}
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = nil
(*temporary) = "attempt to index field 'color' (a nil value)"
buttonText = <unnamed> {
0 = <userdata>
}
db = <table> {
general = <table> {
}
modules = <table> {
}
}
```

**Comment by Vegeir** [comment]
I tested this again on a 100 Warlock.  By default, a new bar profile was created when I logged in for the firs time after installing 2.03.  Here are my observations:

- With bar on bottom, I was able to left and right click the Specialization module with no errors.  I had to move it based on my UI set-up, which has my bars and threat / chat windows hugging the lower edge of my screen.

- Moved bar to top and immediately had the same problem as I initially described above.

- With bar on top, the spec drop-down options are visible, although the opacity is so faint that you can barely read it.  I don't recall 1.10 being that faint.
![lock_top](https://cloud.githubusercontent.com/assets/21963210/18222067/67801faa-7159-11e6-86be-3c0e726ce67e.png)

- Moved bar back to bottom and the drop-down menu options do just that - move downwards off the screen.
![lock_bottom](https://cloud.githubusercontent.com/assets/21963210/18222077/af94dc0e-7159-11e6-8959-60ca958eeca7.png)

- Bugsack collects the errors on mouse-over for the drop-down for specialization.  The functionality to switch specs still works regardless.
 
Hope this helps.  Let me know if I can test further.

**Comment by Vegeir** [comment]
Here is a visual for how the bar performs when initially loaded on a toon which hasn't used it yet.  This is the default profile on my 101 Paladin.  It is set to BOTTOM by default and the "drop down" menu for specialization grow upwards as expected.  The opacity / alpha looks what I would expect it to.  If the bar is moved to TOP and then back to BOTTOM, the problem shown in my second screen capture above reoccurs.

![pally_2016-09-02_2215](https://cloud.githubusercontent.com/assets/21963210/18222133/1c51c5d6-715b-11e6-8591-dffbc36e9cd0.png)

**Comment by Vegeir** [comment]
**TL;DR**: The error only occurs the first time you move the bar and is only present until after re-logging.



Was on my 100 warrior and noticed that after changing to the profile I have set (which has the bar on top) and re-logging that the error did not reoccur.  I switched back to my paladin to confirm and noticed that I could switch spec / loot spec as intended.  The alpha / opacity looks good again as well.

Here is a pic:
![pally_speclooksgood](https://cloud.githubusercontent.com/assets/21963210/18226409/fdeb5860-71d6-11e6-8015-af317c6e4577.png)
